### PR TITLE
End-to-end testing with Jest, Puppeteer

### PR DESCRIPTION
**Pull request by hstove**
This PR includes the happy flow test, as well as a working runner on CircleCI.

- Setup headless browser testing with Puppeteer using Chrome
- Adds a demo app from which to launch the Connect modal
- Tests cross-origin messaging
- Runs the tests with CircleCI

### CODE REVIEW

**Comment by kyranjamie** [comment]
@hstove I accidentally rebased this onto `master`. Undone but commit hashes will have changed, so will need to reset hard onto the remote.

**Comment by kyranjamie** [comment]
8bd062b adds a test for validating the seed phrase has been entered correctly. Currently, there is no handling on this, so the test if failing.

File: tests/ts/e2e/demo.page.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export class DemoPageObject {
+  url = 'http://localhost:3001';
+
+  $openAuthButton = '#auth-action';
+
+  async goToPage() {
+    return page.goto(this.url, { waitUntil: 'domcontentloaded' });
+  }
+
+  async goToFreshPage() {
```
Comment (hstove) [inline tests/ts/e2e/demo.page.ts]: Should this be removed now that we have the fresh page changes?

**Comment by hstove** [comment]
> 8bd062b adds a test for validating the seed phrase has been entered correctly. Currently, there is no handling on this, so the test if failing.

All good, thanks for the heads up. In my opinion, it's fine (and good) to rebase master if only one person is working on a branch at a time, which is true in this case.

**Review by kyranjamie — APPROVED** [review]

**Comment by kyranjamie** [comment]
These tests have broken a few times owing to changes coming in on connect and feature/onboarding. Merging now in a working state so they're not isolated from the main branch.

**Pull request merged**